client/src/modules/shopping-list/index.jsx

```diff
@@ -26,7 +26,10 @@ class ShoppingList extends Component {
   };
 
   componentDidMount() {
-    this.fetchData();
+    const { list } = this.props;
+    if (!list || (list && !list.isArchived)) {
```

Comment:
i would create  method checkIfArchived which returns true or false and then use it here in the if condition, cause it it took me few minutes to understand what's all about